src/softhsm.rs

```diff
@@ -0,0 +1,141 @@
+
+use anyhow::{anyhow, Result};
+use tempdir::TempDir;
+use std::process::Command;
+
+use std::{thread, time};
+
+const TEMPDIR_PREFIX: &str = "ocicrypt-rs";
+
+pub struct SoftHSMSetup {
+    pub statedir_path: String,
+    pub statedir_folder: TempDir,
```

Comment:
maybe the String can be deduced from the TempDir (by adding a method `get_statedir_path(&self)` ? Why do we need both in the struct?